src/dynsight/_internal/vision/detect.py

```diff
@@ -0,0 +1,782 @@
+                image_save_path = images_val_dir / f"{i}.png"
+                label_save_path = labels_val_dir / f"{i}.txt"
+
+            collage.save(image_save_path)
+            with label_save_path.open("w") as f:
+                for line in label_lines:
+                    f.write(line + "\n")
+
+        # Generate the config file for the dataset created
+        self._add_or_create_yaml(self.syn_dataset_path)
+
+    # Just a bridge to the YOLO library
+    def train(
+        self,
+        yaml_file: pathlib.Path,
+        initial_model: str | pathlib.Path = "yolo12x.pt",
+        training_name: str | None = None,
+        training_epochs: int = 100,
+        training_patience: int = 100,
+        batch_size: int = 16,
+        workers: int = 8,
+        device: int | str | list[int] | None = None,
+    ) -> None:
+        """Train a YOLO model on the generated synthetic dataset.
+
+        Args:
+            yaml_file: Path to the dataset YAML configuration file.
+            initial_model: Pretrained model checkpoint to fine-tune.
+            training_name: Name for the training run.
+            training_epochs: Maximum number of training epochs
+            training_patience: Early stopping patience
+                (number of epochs without improvement).
+            batch_size: Batch size for training.
+            workers: Number of dataloader worker threads.
+            device: Device(s) on which to run training.
+        """
+        model = YOLO(initial_model)
+        model.train(
+            data=yaml_file,
+            epochs=training_epochs,
+            patience=training_patience,
+            batch=batch_size,
+            imgsz=self.video_size,
+            workers=workers,
+            project=self.models_path,
+            name=training_name,
+            device=device,
+            plots=False,
+        )
+
+    # Just a bridge to the YOLO library
+    def predict(
+        self,
+        model_path: str | pathlib.Path,
+        detections_iou: float = 0.1,
+        prediction_name: str = "prediction",
+    ) -> None:
+        """Perform object detection predictions on the extracted frames.
+
+        Args:
+            model_path: Path to the trained model weights.
+            detections_iou: IOU threshold for object detection filtering.
+            prediction_name: Name under which to save the prediction results.
+        """
+        model = YOLO(model_path)
+        model.predict(
+            project=self.project_folder,
+            source=self.frames_dir / "0.png",
+            name=prediction_name,
+            augment=True,
+            line_width=2,
+            save=True,
+            show_labels=False,
+            save_txt=True,
+            save_conf=True,
+            iou=detections_iou,
+            max_det=20000,
+        )
+
+    def fit(
+        self,
+        initial_dataset: pathlib.Path,
+        initial_model: str | pathlib.Path = "yolo12x.pt",
+        max_sessions: int = 2,
```

Comment:
Ensure these are safe, usable defaults - i.e., what you would actually use in production